### PR TITLE
Use helm version from github tags. Fix #170

### LINKED ISSUE #170: Upgrade `helm` binary to the latest version
Currently (6.0.0) by default `helm` **3.2.0** is used by plugin.

Please upgrade to the latest version - **3.8.1**

Please consider to use GitHub Api to automatically detect latest version.

```https://api.github.com/repos/helm/helm/releases/latest```

### PR DESCRIPTION
**Pull request by sschnabe**
@slawekjaranowski  FYI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,13 +53,14 @@ The default setting is to construct the Helm download URL based upon the detecte
...
         <chartDirectory>${project.basedir}</chartDirectory>
         <chartVersion>${project.version}</chartVersion>
-        <helmVersion>3.2.0</helmVersion>
       </configuration>
     </plugin>
   </plugins>
 </build>
 ```
 
+If you leave `helmVersion` and `helmDownloadUrl` empty the plugin will determine the latest version based on [https://api.github.com/repos/helm/helm/releases/lates].
```
Comment (axdotl) [inline README.md:62]:
```suggestion
If you leave `helmVersion` and `helmDownloadUrl` empty the plugin will determine the latest version based on <https://api.github.com/repos/helm/helm/releases/latest>.
```

**Comment by slawekjaranowski** [comment]
Thanks for it. 

Issue is being a little more complicated. You must cache response form GitHub api. 
I think that you can create a file in local .m2 repository (`mavenSession.getSettings().getLocalRepository()`) with the latest version and call GitHub api eg. once a day. 

GitHub api request is limited for unauthenticated call up to 60 request per hour per ip.
So for big organization behind firewall can be quickly exceeded.
https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting

You can also use `conditional requests` to minimize rate limit. 
https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests

It is also recommended put your application name in `User-Agent` header.
https://docs.github.com/en/rest/overview/resources-in-the-rest-api#user-agent-required

**Comment by sschnabe** [comment]
I prefer `${java.io.tmpdir}/helm-maven-plugin/version`. What is the advantage of `${HOME}/.m2`?

I'll add UserAgent as property.

**Comment by slawekjaranowski** [comment]
`tmpdir` can be different by executions, especially on CI system
eventually new property with cache path with default value, if someone has own requirement - can changed it

**Comment by sschnabe** [comment]
I've added UserAgent and a simple cache. The cache is using `${java.io.tmpdir}` as default, but can be changed.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**